src/main/java/net/dv8tion/jda/internal/entities/TemplateImpl.java

```diff
@@ -0,0 +1,258 @@
+        this.uses = uses;
+        this.creator = creator;
+        this.createdAt = createdAt;
+        this.updatedAt = updatedAt;
+        this.guild = guild;
+        this.synced = synced;
+    }
+
+    public static RestAction<Template> resolve(final JDA api, final String code)
+    {
+        Checks.notNull(code, "code");
+        Checks.notNull(api, "api");
+
+        Route.CompiledRoute route = Route.Templates.GET_TEMPLATE.compile(code);
+
+        JDAImpl jda = (JDAImpl) api;
+        return new RestActionImpl<>(api, route, (response, request) ->
+                jda.getEntityBuilder().createTemplate(response.getObject()));
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Template> sync()
+    {
+        checkInteraction();
+
+        final Route.CompiledRoute route = Route.Templates.SYNC_TEMPLATE.compile(guild.getId(), this.code);
+
+        JDAImpl jda = (JDAImpl) api;
+        return new RestActionImpl<>(api, route, (response, request) ->
+                jda.getEntityBuilder().createTemplate(response.getObject()));
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> delete()
+    {
+        checkInteraction();
+
+        final Route.CompiledRoute route = Route.Templates.DELETE_TEMPLATE.compile(guild.getId(), this.code);
+
+        return new RestActionImpl<>(api, route);
+    }
+
+    @Nonnull
+    @Override
+    public String getCode()
+    {
+        return this.code;
+    }
+
+    @Nonnull
+    @Override
+    public String getName()
+    {
+        return this.name;
+    }
+
+    @Nullable
+    @Override
+    public String getDescription()
+    {
+        return this.description;
+    }
+
+    @Override
+    public int getUses()
+    {
+        return this.uses;
+    }
+
+    @Nonnull
+    @Override
+    public User getCreator()
+    {
+        return this.creator;
+    }
+
+    @Nonnull
+    @Override
+    public OffsetDateTime getTimeCreated()
+    {
+        return this.createdAt;
+    }
+
+    @Nonnull
+    @Override
+    public OffsetDateTime getTimeUpdated()
+    {
+        return this.updatedAt;
+    }
+
+    @Nonnull
+    @Override
+    public Template.Guild getGuild()
+    {
+        return this.guild;
+    }
+
+    @Override
+    public boolean isSynced()
+    {
+        return this.synced;
+    }
+
+    @Nonnull
+    @Override
+    public TemplateManager getManager()
+    {
+        TemplateManager mng = manager;
+        if (mng == null)
+        {
+            mng = MiscUtil.locked(mngLock, () ->
+            {
+                if (manager == null)
+                    manager = new TemplateManagerImpl(this);
+                return manager;
+            });
+        }
+        return mng;
+    }
+
+    private void checkInteraction()
+    {
+        final net.dv8tion.jda.api.entities.Guild guild = this.api.getGuildById(this.guild.getIdLong());
+
+        if (guild == null)
+            throw new IllegalStateException("Cannot interact with a template without shared guild");
```

Comment:
What about sharding? It could just be that the bot shares a guild on another shard, why is this being restricted here?

Reply:
then i'm not sure how to check this, as it's checked the same in Emote(Manager)Impl too.